product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCase.java

```diff
@@ -128,11 +132,19 @@ public void init() throws Exception {
         File swaggerFile = new File(swaggerFilePath);
         String swaggerContent = readFromFile(swaggerFile.getAbsolutePath());
         JSONObject swaggerJson = new JSONObject(swaggerContent);
-        try {
-            apiPublisher.developSampleAPI(swaggerJson, SUPER_USER, backendEndPoint, true, apiVisibility);
-        } catch (Exception ex) {
-            log.error("API publication failed", ex);
-        }
+        String apiContext = swaggerJson.get("basePath").toString();
+
+        APIRequest apiRequest = new APIRequest(apiName, apiContext, apiVisibility, apiVersion, "", "Gold",
+                new URL(backendEndPoint));
+        HttpResponse serviceResponse = apiPublisher.addAPI(apiRequest);
+        verifyResponse(serviceResponse);
+
+        APILifeCycleStateRequest updateRequest =
+                new APILifeCycleStateRequest(apiName, SUPER_USER,
+                        APILifeCycleState.PUBLISHED);
+        serviceResponse = apiPublisher.changeAPILifeCycleStatus(updateRequest);
+        verifyResponse(serviceResponse);
+
```

Comment:
after publishing API we need to wait for the deployment

Reply:
Fixed with 9fd069d5b8dea92d9947914e938a26f21332765f